src/_ctf-private/ctf-consent-management/useCtfConsent.ts

```diff
@@ -0,0 +1,22 @@
+import { BaseConfig, createConsentManager } from '@contentful/experience-consent-manager';
+import { useQuery } from '@tanstack/react-query';
+
+export const useCtfConsent = (props?: Pick<BaseConfig, 'onConsentChanged' | 'onError'>) =>
+  useQuery({
+    queryKey: ['osano-content'],
+    queryFn: async () => {
+      const manager = createConsentManager({
+        appKey: process.env.NEXT_PUBLIC_OSANO_APP_KEY || '',
+        osano: {
+          configId: process.env.NEXT_PUBLIC_OSANO_CONFIG_ID || '',
+        },
```

Comment:
Should we handle the errors when the keys are missing? I was tried and this is what happens. If one follows the docs, it should be sufficient.

<img width="640" alt="Screenshot 2023-02-16 at 15 22 02" src="https://user-images.githubusercontent.com/103024358/219390587-054c6c06-971f-4838-ba1f-8a10d4bd6672.png">

Are we error handling the `useCtfConsent` the existence of those? What would happen if they are not defined?

Reply:
It shouldn't ever be an issue for our users as the key _should_ always be there, but I'll have a look if we can put a fix on this

Reply:
That is true, this is a non-blocking comment, feel free to merge this work!